loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/LoadFlowResult.java

```diff
@@ -14,19 +15,56 @@
  * Loadflow result API.
  *
  * @author Geoffroy Jamgotchian {@literal <geoffroy.jamgotchian at rte-france.com>}
+ * @author Damien Jeandemange {@literal <damien.jeandemange at artelys.com>}
  */
 public interface LoadFlowResult {
 
+    /**
+     * Result for one slack bus
+     */
+    interface SlackResult {
+        /**
+         * Get the slack bus id.
+         * @return the slack bus id
+         */
+        String getBusId();
+
+        /**
+         * Get slack bus active power mismatch in MW.
+         * @return the slack bus active power mismatch in MW
+         */
+        double getBusActivePowerMismatch();
+    }
+
     /**
      * Result for one component
      */
     interface ComponentResult {
 
+        /**
+         * Calculation status
+         */
         enum Status {
+            /**
+             * Loadflow converged
+             */
             CONVERGED,
+            /**
+             * Non-convergence by iteration limit
+             */
             MAX_ITERATION_REACHED,
+            /**
+             * Non-convergence by solver failure
+             */
             SOLVER_FAILED,
-            FAILED
+            /**
+             * Non-convergence
+             */
+            FAILED,
+            /**
+             * Component was not calculated, e.g. component does not have any PV node
+             */
+            NOT_CALCULATED
```

Comment:
In open load flow, we have for NR status ` NO_CALCULATION`.

Reply:
Will change for NO_CALCULATION. Also realizing javadoc is wrong, no PV node should be failed. NO_CALCULATION should be for not-energized components, e.g. no generation. Component with generation but without PV node should be flagged FAILED.

UNREALISTIC_STATE is in my opinion OpenLoadFlow specific, and should translate to just FAILED. Also I am not so comfortable with the SOLVER_FAILED status, which is implementation specific. In LoadFlow API I would just leave FAILED.

Reply:
Just a question: what is exactly "No **more** PV bus"? No PV bus at beginning or during the load flow calculation?
What is the exact criteria of a de-energized component for you?

Reply:
Agree with the text status, great idea.

Reply:
yes, good idea to remove the too specific solver failed and add a text to give more details

Reply:
> Just a question: what is exactly "No **more** PV bus"? No PV bus at beginning or during the load flow calculation? What is the exact criteria of a de-energized component for you?

Bad example ... Should be "No PV bus".

de-energized is a component without any connected equipment with voltage support. In this case Load Flow implementations shoud report NO_CALCULATION. The component is dead / not viable.

If at start of calculation or if during calculation all equipments with voltage support are PQ or change to PQ due to reactive limits -> Component should be flagged FAILED